src/Npgsql/PostgresEnvironment.cs

```diff
@@ -22,6 +23,43 @@ static class PostgresEnvironment
 
         public static string? TimeZone => Environment.GetEnvironmentVariable("PGTZ");
 
+        public static string? Options => Environment.GetEnvironmentVariable("PGOPTIONS");
+
+        public static Dictionary<string, string> ParsedOptions
+        {
+            get
+            {
+                if (_optionsCache == Options)
+                    return ParsedOptionsCache;
+
+                _optionsCache = Options;
+                ParseOptions();
+                return ParsedOptionsCache;
+            }
+        }
+
+        static readonly Dictionary<string, string> ParsedOptionsCache = new Dictionary<string, string>();
+
+        static string? _optionsCache;
+
+        static void ParseOptions()
```

Comment:
There's a (rare) race condition here if two physical connections connect at the same time, and modify the dictionary concurrently.

To manage this properly, this would either:
* Acquire a lock around the parsing operation, or
* Not operate directly on ParseOptionsCache, but rather on a new dictionary instance in the stack, which would at the end be assigned to ParsedOptionsCache.

FWIW it may not be critical to cache here - establishing a physical connection is quite a heavy thing, and compared to the initial handshake parsing this environment variable is likely to be negligible. So I'm fine with removing this altogether - but if we keep it, it should be safe.

Reply:
Yeah, I think so too. I removed the caching of the parsed options. We can always re-enable it if we identify this as a relevant bottleneck or want to squeeze out the last bits of physical connection startup delay for rare configuration variants. ;-)